keadm/cmd/keadm/app/cmd/debug/check.go

```diff
@@ -0,0 +1,374 @@
+}
+
+func CheckCPU() error {
+	percent, err := cpu.Percent(time.Second, false)
+	if err != nil {
+		return err
+	}
+
+	cpuNum, err := cpu.Counts(true)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("CPU total: %v core, Allowed > %v core\n", cpuNum, constant.AllowedValueCPU)
+	fmt.Printf("CPU usage rate: %.2f, Allowed rate < %v\n", percent[0]/100, constant.AllowedCurrentValueCPURate)
+
+	if cpuNum < constant.AllowedValueCPU || percent[0]/100 > constant.AllowedCurrentValueCPURate {
+		return errors.New("cpu check failed")
+	}
+	return nil
+}
+
+func CheckMemory() error {
+	mem, err := mem.VirtualMemory()
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Memory total: %.2f MB, Allowed > %v MB\n", float32(mem.Total)/constant.MB, constant.AllowedValueMemory/constant.MB)
+	fmt.Printf("Memory Free total: %.2f MB, Allowed > %v MB\n", float32(mem.Free)/constant.MB, constant.AllowedCurrentValueMem/constant.MB)
+	fmt.Printf("Memory usage rate: %.2f, Allowed rate < %v\n", mem.UsedPercent/100,
+		constant.AllowedCurrentValueMemRate)
+
+	if mem.Total < constant.AllowedValueMemory ||
+		mem.Free < constant.AllowedCurrentValueMem ||
+		mem.UsedPercent/100 > constant.AllowedCurrentValueMemRate {
+		return errors.New("memory check failed")
+	}
+
+	return nil
+}
+
+func CheckDisk() error {
+	parts, err := disk.Partitions(false)
+	if err != nil {
+		return err
+	}
+
+	diskInfo, err := disk.Usage(parts[0].Mountpoint)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Disk total: %.2f MB, Allowed > %v MB\n", float32(diskInfo.Total)/constant.MB, constant.AllowedValueDisk/constant.MB)
+	fmt.Printf("Disk Free total: %.2f MB, Allowed > %vMB\n", float32(diskInfo.Free)/constant.MB, constant.AllowedCurrentValueDisk/constant.MB)
+	fmt.Printf("Disk usage rate: %.2f, Allowed rate < %v\n", diskInfo.UsedPercent/100, constant.AllowedCurrentValueDiskRate)
+
+	if diskInfo.Total < constant.AllowedValueDisk ||
+		diskInfo.Free < constant.AllowedCurrentValueDisk ||
+		diskInfo.UsedPercent/100 > constant.AllowedCurrentValueDiskRate {
+		return errors.New("disk check failed")
+	}
+
+	return nil
+}
+
+func CheckDNS(domain string) error {
+	r, err := net.LookupHost(domain)
+	if err != nil {
+		return fmt.Errorf("dns resolution failed, domain: %s err: %s", domain, err)
+	}
+	if len(r) > 0 {
+		fmt.Printf("dns resolution success, domain: %s ip: %s\n", domain, r[0])
+	} else {
+		fmt.Printf("dns resolution success, domain: %s ip: null\n", domain)
+	}
+	return err
+}
+
+func CheckDNSSpecify(domain string, dns string) error {
+	if dns != "" {
+		net.DefaultResolver = &net.Resolver{
+			PreferGo: true,
+			Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+				d := net.Dialer{
+					Timeout: time.Millisecond * time.Duration(4000),
+				}
+				return d.DialContext(ctx, "udp", fmt.Sprintf("%s:53", dns))
+			},
+		}
+	}
+	return CheckDNS(domain)
+}
+
+func CheckNetWork(IP string, timeout int, edgeHubURL string) error {
+	if IP == "" && edgeHubURL == "" {
```

Comment:
why `edgeHubURL == “”`？

Reply:
This should be cloudhubUrl. If neither ip nor url is provided, then use the default ip for testing